lib/src/main/kotlin/app/cash/kfsm/Transition.kt

```diff
@@ -0,0 +1,18 @@
+package app.cash.kfsm
+
+import app.cash.quiver.extensions.ErrorOr
+import arrow.core.right
+
+open class Transition<S : State>(val from: Set<S>, val to: S) {
+
+  init {
+    require(from.isNotEmpty()) { "from must not be empty" }
```

Comment:
It's unfortunate that this is a runtime check for something that is knowable at compile time (using a NEL)

Reply:
updated